imap_processing/hi/l1c/hi_l1c.py

```diff
@@ -8,22 +8,35 @@
 import numpy as np
 import pandas as pd
 import xarray as xr
+from numpy import typing as npt
 
 from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
 from imap_processing.cdf.utils import parse_filename_like
-from imap_processing.hi.l1a.science_direct_event import DE_CLOCK_TICK_S
+from imap_processing.hi.l1a.science_direct_event import (
+    DE_CLOCK_TICK_S,
+    HALF_CLOCK_TICK_S,
+)
 from imap_processing.hi.utils import (
     CoincidenceBitmap,
     create_dataset_variables,
     full_dataarray,
+    parse_sensor_number,
 )
 from imap_processing.spice.geometry import (
     SpiceFrame,
     frame_transform,
     frame_transform_az_el,
 )
+from imap_processing.spice.spin import (
+    get_instrument_spin_phase,
+    get_spin_data,
+)
 from imap_processing.spice.time import ttj2000ns_to_et
 
+N_SPIN_BINS = 3600
+SPIN_PHASE_BIN_EDGES = np.linspace(0, 360, N_SPIN_BINS + 1) / 360
+SPIN_PHASE_BIN_CENTERS = SPIN_PHASE_BIN_EDGES[:-1] + SPIN_PHASE_BIN_EDGES[1] / 2
```

Comment:
I think this may be wrong, or at least confusing to me at first glance... Sorry if I typed it out wrong and you used the suggested change feature!

```suggestion
SPIN_PHASE_BIN_CENTERS = (SPIN_PHASE_BIN_EDGES[:-1] + SPIN_PHASE_BIN_EDGES[1:]) / 2
```

Reply:
Oh, I see. You want to take the mean of the two edges for each bin. That is a more robust way. I was counting on the fact that `SPIN_PHASE_BIN_EDGES[1]` defines the width of a single bin since `SPIN_PHASE_BIN_EDGES[0] == 0`.